src/sentry/issues/attributes.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from enum import Enum
+from typing import Optional
+
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+from sentry.models import Group, GroupOwner
+from sentry.signals import issue_assigned, issue_deleted, issue_unassigned
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+class Operation(Enum):
+    CREATED = "created"
+    UPDATED = "updated"
+    DELETED = "deleted"
+
+
+def _log_group_attributes_changed(
+    operation: Operation,
+    model_inducing_snapshot: str,
+    column_inducing_snapshot: Optional[str] = None,
+) -> None:
+    metrics.incr(
+        "group_attributes.changed",
+        tags={
+            "operation": operation.value,
+            "model": model_inducing_snapshot,
+            "column": column_inducing_snapshot,
+        },
+    )
+
+
+@receiver(
+    post_save, sender=Group, dispatch_uid="post_save_log_group_attributes_changed", weak=False
+)
+def post_save_log_group_attributes_changed(
+    instance, sender, created, update_fields, *args, **kwargs
+):
+    try:
+        if created:
+            _log_group_attributes_changed(Operation.CREATED, "group", None)
+        else:
+            # we have no guarantees update_fields is used everywhere save() is called
+            # we'll need to assume any of the attributes are updated in that case
+            attributes_updated = {"status", "substatus", "first_seen", "num_comments"}.intersection(
+                update_fields or ()
+            )
+            if attributes_updated:
+                _log_group_attributes_changed(
+                    Operation.UPDATED, "group", "-".join(sorted(attributes_updated))
+                )
+            else:
+                _log_group_attributes_changed(Operation.UPDATED, "group", "unknown")
```

Comment:
If someone does `update(fields=["last_seen"])` (as an example) we'd log that as an unknown update, but we should probably skip it?

Reply:
Hmm, true. I was kinda being overly aggressive with this approach to log a metric in the case when we don't know whether the fields we cared about were actually updated. But I think we should probably be ok since these fields are rarely updated. `status`, and `substatus` are explicitly updated but I believe we took care to explicitly fire the `post_save` when those fields are updated as part of issue states.